nixos/modules/programs/x2goserver.nix

```diff
@@ -0,0 +1,121 @@
+    superenicer = { "enable" = "no"; };
+    telekinesis = { "enable" = "no"; };
+  };
+  confText = generators.toINI {} (recursiveUpdate defaults cfg.config);
+  x2goServerConf = pkgs.writeText "x2goserver.conf" confText;
+
+  x2goAgentOptions = pkgs.writeText "x2goagent.options" ''
+    X2GO_NXOPTIONS=""
+    X2GO_NXAGENT_DEFAULT_OPTIONS="${concatStringsSep " " cfg.nxagentDefaultOptions}"
+  '';
+
+in {
+  options.programs.x2goserver = {
+    enable = mkEnableOption "x2goserver";
+
+    logLevel = mkOption {
+      type = types.str;
+      default = "notice";
+      description = ''
+        The log level x2goserver will log at.
+        Possible levels are: emerg, alert, crit, err, warning, notice, info, debug
+      '';
+    };
+
+    nxagentDefaultOptions = mkOption {
+      type = types.listOf types.str;
+      default = [ "-extension GLX" "-nolisten tcp" ];
+      example = [ "-extension GLX" "-nolisten tcp" ];
+      description = ''
+        List of default nx agent options.
+      '';
+    };
+
+    config = mkOption {
+      type = types.attrsOf types.attrs;
+      default = {};
+      description = "x2goserver.conf ini configuration as nix attributes.";
+      example = literalExample ''
+        superenicer = {
+          "enable" = "yes";
+          "idle-nice-level" = 19;
+        };
+        telekinesis = { "enable" = "no"; };
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.groups.x2go = {};
+    users.users.x2go = {
+      createHome = true;
+      home = "/var/lib/x2go";
+      group = "x2go";
+    };
+
+    security.wrappers.x2gosqliteWrapper = {
+      source = "${pkgs.x2goserver}/lib/x2go/libx2go-server-db-sqlite3-wrapper.pl";
+      owner = "root";
+      group = "x2go";
+      setgid = true;
+    };
+    security.wrappers.x2goprintWrapper = {
+      source = "${pkgs.x2goserver}/bin/x2goprint";
+      owner = "root";
+      group = "x2go";
+      setgid = true;
+    };
+    # x2goclient sends SSH commands with preset PATH set to
+    # "/usr/local/bin;/usr/bin;/bin". Since we cannot filter arbitrary ssh
+    # commands, we have to make the following executables available.
+    system.activationScripts.x2goserver = with pkgs; ''
+      mkdir -p /usr/local/bin
+      for exe in ${x2goserver}/bin/x2go*
+      do
+        ln -sf $exe /usr/local/bin/
+      done
+      for util in \
+        ${xorg.xrandr}/bin/xrandr ${xorg.xmodmap}/bin/xmodmap \
+        ${utillinux}/bin/setsid ${gawk}/bin/awk ${gnused}/bin/sed \
+        ${coreutils}/bin/cp ${coreutils}/bin/chmod
+      do
+        ln -sf $util /usr/local/bin/
+      done
+      mkdir -p /var/lib/x2go/conf
+      if [ ! -f /var/lib/x2go/x2go_sessions ]
+      then
+        ${x2goserver}/bin/x2godbadmin --createdb
+      fi
+      cp -r ${x2goserver}/etc/x2go/* /var/lib/x2go/conf/
+      ln -sf ${x2goServerConf} /var/lib/x2go/conf/x2goserver.conf
+      ln -sf ${x2goAgentOptions} /var/lib/x2go/conf/x2goagent.options
+      chmod o+rx /var/lib/x2go
```

Comment:
Use the declarative `systemd.tmpfiles` (`man tmpfiles.d`) to set up /var/lib instead

Reply:
This is actually a fix for a chmod in `--createdb` earlier. I'll change the home directory (which is where the sqlite db is put) to `/var/lib/x2go/db` and put `/var/lib/x2go` in tmpfiles.